### PR TITLE
[WIP] Show lists of topics within a stream

### PR DESCRIPTION
**Pull request by TomaszKolek**

### CODE REVIEW

**Comment by smarx** [comment]
*Automated message from Dropbox CLA bot*

@TomaszKolek, it looks like you've already signed the Dropbox CLA. Thanks!

**Comment by TomaszKolek** [comment]
http://imgur.com/a/F7xbc
This is how it looks right now,
I'd love to get some feedback from architecture of the project point of view.

**Comment by timabbott** [comment]
I think visually we want the topics to be (1) indented, so they're clearly under the stream and (2) have a row you can click on to narrow to the stream+topic pair.  

UI-wise, I think we want the model we have on web where we show up to 5 initially, with a "more topics" button to show more.

**Comment by TomaszKolek** [comment]
@timabbott how show more topic in details? 
when you click show more topics  you see a few topics but when you scroll up on messages part you're able to see more topics - what is the logic behind that - I mean correlations between scrolling on messages panel and the number of topics

**Comment by timabbott** [comment]
Yeah, so the webapp hasn't finished being ported to use the endpoint I recommended here to fetch all topics from the backend.  It's on @showell's TODO list :)

**Comment by TomaszKolek** [comment]
@borisyankov would love if you could take a look from code point of view and tell me what should be moved where and why ;-) I'm pretty sure we've got some conventions about logical component and "visual" component.

**Comment by borisyankov** [comment]
Mobile apps does not copy web apps, for good reason. We should absolutely not do it the same way as in the web app.
There are many limitations:
* the screen is 1/2nd or 1/3rd vertically of what the web is
* also the items themselves are at least twice as big, so they can be touchable
* we need to test on minimal phone sizes, which are iPhone 5 ands some random Android 4".

There is no space for the topics there at all

**Comment by TomaszKolek** [comment]
Maybe somehow limited ScrollView (I mean space for 5-6 topics that can be scrolled) ?

**Comment by timabbott** [comment]
What I was imagining was if there are more than 5, you click the "more topics" button and it takes you to a view that only shows topics in that stream with a scrollview to see them all.  Could make sense to go directly to that sort of view, but I think (1) we need to have an easy way to see and choose the topics and (2) we probably don't want to make it impossible to then scroll down to the next stream.

**Comment by trueskawka** [comment]
I'm wondering if this should be visible from the home view - maybe if someone clicks a stream, then it narrows and shows the list of topics in the stream instead of stream list like it does in the web app? I agree that showing topics in the home narrow may be a lot to handle :)

**Comment by neerajwahi** [comment]
I would add a light chevron next to each stream in the left sidebar. Clicking on the chevron exposes the most recent 5 topics for the stream. Clicking on the stream name itself narrows like it does now.

**Comment by timabbott** [comment]
I like that design.

**Comment by TomaszKolek** [comment]
Will try to prepare something like that.

**Comment by neerajwahi** [comment]
@TomaszKolek are you planning to open a new PR for this stuff (and use this one as a request for comments)? If so I'll close this one

**Comment by TomaszKolek** [comment]
Please keep it as it is - I will modify those commits

**Comment by neerajwahi** [comment]
Cool. In that case would you mind changing the title to something like 'Add topics to stream sidebar'?

**Comment by TomaszKolek** [comment]
I'm getting back to this. So the first iteration of the feature should only show 5 recent topics?

**Comment by neerajwahi** [comment]
@TomaszKolek sorry missed your earlier comment. Yeah 5 topics for now but we should make it configurable

File: src/api/getTopicsOfStream.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+export default async (
+  auth,
+  streamId
+) =>
+  apiGet(
+    auth,
+    `users/me/${streamId}/topics`,
+    {},
+    res => res.messages
+  );
```
Comment (borisyankov) [inline src/api/getTopicsOfStream.js]: You can do this function as a single, separate commit.

**Comment by borisyankov** [comment]
@TomaszKolek we still need this work at some point, but the PR is a bit outdated at this point, so I will close it.
Feel free to use part of the code later if you decide to work on it - for example the topic API call is fine even as a stand-alone PR. 

The list will need to be implemented with FlatList and the styling needs to be gotten via context (nothing complicated, we can chat about it when you want to work on it)

**Pull request closed without merging**